dirsync/syncer.py

```diff
@@ -158,9 +158,9 @@ def _compare(self, dir1, dir2):
 
                 if add_path:
                     left.add(path)
-                    anc_dirs = re_path[:-1].split('/')
+                    anc_dirs = re_path.split('/')
```

Comment:
I really can't remember why the [:-1] was added here. This dates from when I completely rewrote / adapted dirsync from the original author's code. It was possibly to remove the trailing slash for directories, and I may never have hit this line with `re_path` being a file. However, to safe-guard against adding too many items to `left`, shouldn't we either use `re_path.rstrip("/").split("/")` or add a `if not ad: continue` in the for loop that follows?